assets/src/date/getDateObjectWithTimezone.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import moment from 'moment-timezone';
+
+// return a moment date object based on dateFormatting.timezone if one is present
+// if no timezone check for gmtOffset, if that is present use it to set timezone based on UTC
+// by default return moment date as is based on browser time
+
+/**
+ * @summary                        Get a date object of the time right now in any timezone
+ * Regardless of timezone or offset we want to have the same starting point of time.
+ * We need to specify moment to be created in UTC rather than browser timezone.
+ * Any moment created with moment.utc() will be in UTC mode, and any moment created with moment() will not.
+ * @param {Object} dateFormatting  Object responsible for relevant timezone manipulation based off an UTC date in Moment.
+ * Here we are looking for 1) dateFormatting.timezone, if no timezone 2) check for gmtOffset, if no gmtOffset use UTC
+ * @return {Date}                  Date object formatted to timezone specified in param, UTC by default
+ */
+
+export function getDateObjectWithTimezone(dateFormatting = {}) {
+  const { timezone, gmtOffset } = dateFormatting;
```

Comment:
This PR does not include passing GMT offset & timezone info from PHP. Intentional?

Reply:
yes :) I wanted to separate functionality and new date structure to focus on things like tests and names and prevent the PR from growing.